src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -96,10 +99,11 @@ private async Task<bool> IsEnabledWithVariantsAsync<TContext>(string feature, TC
             {
                 isFeatureEnabled = false;
             }
-            else if ((featureDefinition.Variants?.Any() ?? false) && featureDefinition.Allocation != null)
-            {
-                VariantDefinition variantDefinition;
 
+            VariantDefinition variantDefinition = null;
+
```

Comment:
`(featureDefinition.Variants?.Any() ?? false)`

The code works but the uncommonness always delays me a bit. What do you think to change it to a bit longer, but hopefully more digestable

`featureDefinition.Variants != null && featureDefinition.Variants.Any()`

?